src/downloadAzureProject/setupProjectFolder.ts

```diff
@@ -0,0 +1,122 @@
+    const resourceId: string | null = requiredInputs.resourceId;
+    const defaultHostName: string | null = requiredInputs.defaultHostName;
+    const devContainerName: string | null = requiredInputs.devContainerName;
+    const language: string | null = requiredInputs.language;
+    const downloadAppContent: string | null = requiredInputs.downloadAppContent;
+
+    if (resourceId && defaultHostName && devContainerName && language && downloadAppContent) {
+        ext.context.globalState.update(GlobalStates.projectLanguage, getProjectLanguageForLanguage(language));
+        const vsCodeFilePathUri: vscode.Uri = vscode.Uri.file(filePath);
+        const toBeDeletedFolderPathUri: vscode.Uri = vscode.Uri.joinPath(vsCodeFilePathUri, 'temp');
+
+        try {
+            const functionAppName: string = getNameFromId(resourceId);
+            const downloadFilePath: string = vscode.Uri.joinPath(toBeDeletedFolderPathUri, `${functionAppName}.zip`).fsPath;
+
+            vscode.window.showInformationMessage(localize('settingUpFunctionAppLocalProjInfoMessage', `Setting up project for function app '${functionAppName}' with language '${language}'.`));
+
+            if (downloadAppContent === 'true') {
+                // NOTE: We don't want to download app content for compiled languages.
+                await callWithTelemetryAndErrorHandling('azureFunctions.getFunctionAppMasterKeyAndDownloadContent', async (actionContext: IActionContext) => {
+                    const hostKeys: WebSiteManagementModels.HostKeys | undefined = await requestUtils.getFunctionAppMasterKey(resourceId, actionContext);
+                    // tslint:disable-next-line: no-unsafe-any
+                    if (!!hostKeys && hostKeys.masterKey) {
+                        await requestUtils.downloadFunctionAppContent(defaultHostName, downloadFilePath, hostKeys.masterKey);
+                    }
+                });
+            }
+
+            const projectFilePathUri: vscode.Uri = vscode.Uri.joinPath(vsCodeFilePathUri, `${functionAppName}`);
+            const projectFilePath: string = projectFilePathUri.fsPath;
+            const devContainerFolderPathUri: vscode.Uri = vscode.Uri.joinPath(projectFilePathUri, '.devcontainer');
+            ext.context.globalState.update(GlobalStates.projectFilePath, projectFilePathUri.fsPath);
+            await callWithTelemetryAndErrorHandling('azureFunctions.extractContentAndDownloadDevContainerFiles', async (_actionContext: IActionContext) => {
+                if (downloadAppContent === 'true') {
+                    // tslint:disable-next-line: no-unsafe-any
+                    await extract(downloadFilePath, { dir: projectFilePath });
+                }
+                await requestUtils.downloadFile(
+                    `https://raw.githubusercontent.com/microsoft/vscode-dev-containers/master/containers/${devContainerName}/.devcontainer/devcontainer.json`,
```

Comment:
I don't think we can rely on this URL to work going forward. For example, maybe they change the default branch from "master" to "main" (a common thing happening lately) or maybe they change the file structure.

An easy way to fix that would be to just get rid of dev containers stuff. @fiveisprime what do you think - is dev containers worth it for this? A few reasons against it IMO:
1. If the point of using the dev container is to reduce the friction of installing dev tools on my local machine, I'm not sure this does that. For example, if I have a JavaScript or C# project, I probably already have Node.js or .NET on my machine, but it seems less likely that I have Docker. The only remaining tool to install is the func cli, but we already have some logic to help users with that (and we want to keep improving that logic) and it's only necessary if you want to debug 
1. I don't think our users typically use dev containers for local development (might be worth checking telemetry). I'm also pretty sure the dev containers extension is the least used Remote extension (compared to SSH and WSL)

Reply:
@ejizba 
We want to use devContainers for this feature so that customers don't need to install each language specific libraries on their machine.

Also, this feature is a two step process and the first step would be an msi which will install all the requirements like Docker, etc on the customer's machine before they hit this flow. This was mentioned in our initial demo meeting with Matt and you.

For the URL, we are working with Functions team to get an API which would return the devContainer but, till then we would be using this method to unblock the feature.

We can discuss this over a call if you like.

Reply:
@ejizba  and @fiveisprime Do you want me to setup a call to talk about this and some other questions related to our feature that came up in this PR?

Reply:
Discussed in the meeting